File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let left = 0;
    let profit = 0;
    for (let i = 0; i < prices.length; i++) {
        if (prices[left] > prices[i]) {
            left = i;
            continue;
        }
        if (prices[i] - prices[left] > profit) profit = prices[i] - prices[left];
    }
    return profit;
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.rb
# @param {Integer[]} prices
# @return {Integer}
def max_profit(prices)
    currentMin = 0
    leftNumber = prices[0]
    prices.drop(1).each do |price|
        puts "#{price}, #{currentMin}, #{leftNumber}"
        if price - leftNumber > currentMin
            currentMin = price-leftNumber
        else
            leftNumber = price if (price < leftNumber)
        end
    end

    return currentMin
end